### PR TITLE
Copy TE part number 1217861-1.kicad_mod from brunoeagle/kicad-open-mo…

### PR DESCRIPTION
**Pull request by chmorgan**
…dules on github

Verified that this footprint properly fits the part on an actual design.

### CODE REVIEW

**Comment by chmorgan** [comment]
Updated to correct issues caught by klc

**Comment by chmorgan** [comment]
Any updates on merging this footprint in? I have several that I'd like to get back into the mainline libraries but didn't want to push any more until I got at least one through.

**Comment by SchrodingersGat** [comment]
Hi Chris thanks for your contribution :)

We like to have the filename provide some useful information about the footprint (more than just the MPN).

I'm not sure if we have any of these quick-connect tabs in the library yet so we may have to work out how to name them!

In general:

1. Manufacturer name goes first for footprints that are specific to a particular manufacturer
1. Try to include series name
1. Try to include dimensions

I would suggest a name such as:

`TE_Faston_Tab_1217861-1_6.35x0.81mm`

The actual footprint looks good, once we settle on a name I am happy to merge.

**Comment by chmorgan** [comment]
@SchrodingersGat that naming works for me. If I rename as such (after verifying the dimensions) will that work? That particular tab is actually 0.250", should we be using mm, inches, both?

**Comment by SchrodingersGat** [comment]
mm please :)

**Comment by evanshultz** [comment]
Please see https://github.com/KiCad/Connectors.pretty/issues/56 where I proposed a naming convention some time ago for these types of parts. As these are generic, using a specific manufacterer name (TE) and a manufacturer-specific term like `Faston` concerns me a bit.

Also, I created about 50 generic tab footprints some time ago. If we agree on a naming scheme I should get around to submitting a PR for them and also submit the naming scheme to Rene.

**Comment by evanshultz** [comment]
I live in a country which uses stupid Imperial measurements, but I see tabs refereed to in inches all over the world. Does anyone else have a feeling what's more common? If inches is most common, would it be best to use or at least include inches? I suspect for a large part of KiCad's users inches will be most familiar for this type of part even if the users are comfortable with mm for other parts.

**Comment by chmorgan** [comment]
@evanshultz I'm using specific names and terms because these footprints are for those specific parts. While it may be ok to have a generic part I'm not sure anyone building a board would want to hope a footprint worked. Unless of course these things really are more standard than that.

I'll go along with whatever you guys want for a naming convention but would prefer we at least mention the mfg and pn in the footprint or in some list of alternate components.

**Comment by evanshultz** [comment]
I think the footprint name should start with `Tab_` for grouping.

Regarding the generic parts, I do feel they're helpful. KiCad uses mostly generic footprints which are assigned to a symbol in CvPcb. The symbol is specific (often) but then can use mostly generic footprints (with a few specific ones). This connector is more difficult, because there isn't a schematic symbol for 2-pin tabs (not that I see). If so, then that symbol could show this footprint and generic footprints, or possibly just generic ones. In any case, the user would only see footprints that fit that specific symbol. Not having a specific schematic symbol means that a specific footprint is probably best, at least at this time. This is as much for me to find and recall my thoughts later as it is pertinent to this PR.

In addition, we're discussed a re-organization of the connector footprints which may make the organization of this footprint different than it is now. But that shouldn't stop your PR which is great now.

Anyway, a lot of words just to say that this is fine and I'm thankful for for your contributions. But this may not be how the libraries are structured in the future.

**Comment by poeschlr** [comment]
Regarding generic part discussion. I am not sure this is truly a generic part. Even thought it looks very similar to [for example some molex connectors](http://www.molex.com/pdm_docs/sd/197054003_sd.pdf) there are slight differences. This part requires a drill of 1.4mm wheras molex requires 1.7mm The overall height differs as well.

---

My suggestions for FP naming: (from https://github.com/KiCad/kicad-library/issues/1447)
option 1: Standardized connectors with manufacturer specific footprints
`[generic series name]_[manufacturer name]{_[manufacturer series name]}_[manufacturer part number]{_Pad[pad length]{x[pad width]}}_[orientation]{_footprint options}`
Would mean `Tab_TE_Faston_1217861-1_6.35x0.81mm_Vertical` (You might notice that my rule would not result in this name. I smuggled the connector size in without it being mentioned in the rule. I will look into this tomorrow morning.)

---

option 2:
`Tab_[connector_dimension]mm_H[height over board]mm_Drill[drill size]mm{_Pad[pad length]{x[pad width]}}_[orientation]{_footprint options}`
Result: `Tab_6.35x0.81mm_H8.94mm_Drill1.4mm` 
Or if we omit units: `Tab_6.35x0.81_H8.94_Drill1.4`

Edit: this naming scheme assumes the fins are on the same position for all such parts. I am not convinced that this is the case. 

---

Regarding pin numbering:
I would give both pins the same number. Similar to how it is done for bush buttons where two pins are connected. 
That way the symbol that could be used is the generic Conn_01x01. 
But this has the footprint filter `Connector*:*_1x??x*mm`. In other words it expects information about pin configuration which does not make sense for single pin connectors. This is another thing that i will look into tomorrow.

---

Edit: yes i notice that i am a hypocrite for starting a naming convention discussion outside of my own naming convention thread after telling @evanshultz that it is not a good idea. I am truly sorry.

**Comment by poeschlr** [comment]
Another thing:
I think this should be moved to a lib that holds all such connectors.
Maybe Conn_Tab? (Is tab the unique name used for such connectors?)

Edit: I nearly forgot. I would prefer the manufacturer specific naming convention.

---

Now for the footprint:
Courtyard offset is not correct. (should be 0.5mm measured from the pad in x direction and from the fins in y direction.)
Fab outline seems a bit strange (The fins would protrude from this outline in y direction!)
The silk reference should be outside the part. (It is currently inside the fab outline and therefore for sure below the fins and maybe even on top of the pads)
Edit: it might be a good idea to mark out a copper keepout area where the fins touch the pcb. (See KLC on how this is done. I am now on mobile already in bed.)

screenshot from 2017-10-07 02-31-15 png 2017_10_07_02_43_10 0
edit: **sketch is not to scale**

**Comment by SchrodingersGat** [comment]
A quick google search for "tab connector" returns mostly footprints that match this style, so I think that passes the sanity check. `Conn_Tab` or `Conn_Blade_Tab` ? 

If you have a strong preference for either option let me know and I'll make a new repo

**Comment by evanshultz** [comment]
@poeschlr 
1/4W resistors aren't generic, but we have generic footprints. Different vendors have slight dimensional differences on otherwise compatible parts, but they're "close enough" to share a footprint. That's how I see tabs. We can certainly make atomic footprints, but there are several groupings in which 90% of all tabs will fit.

For the symbol, I think we need something new. A symbol that communicates to the user that the connector is a tab. Something like this (which I will finish off and submit as a PR if you guys like it):
image

@SchrodingersGat 
In my experience, `Tab` and `Blade` are interchangeable. So I'd vote for only one.

**Comment by SchrodingersGat** [comment]
https://github.com/KiCad/Connectors_Tab.pretty

Note: The naming is `Connectors_Tab` to stay consistent with the existing connector libraries.

When we merge them together for v5 we will rename them all to `Conn_xxx` instead.

**Comment by SchrodingersGat** [comment]
@evanshultz if there is only one logical connection on the tab, I think you only need one schematic pin.. That symbol to me looks like it is a jumper

**Comment by poeschlr** [comment]
@evanshultz 
A symbol with two pins that have the same number does not work (It is also not necessary. A symbol with a single pin is enough. The schematic designer does not care how many pads there really are.)
I will simply change the footprint filter to `Conn*:*` for the single pin symbol. (for single pin connectors the number of rows is irrelevant.)
I need to change the filters anyway because we decided to name the libs Conn_xx instead of Connector_xx.

Also consider there might be connectors for high current applications that have more than two pads connected to the same connection point. (Would you then make a symbol for each of them? What if such a connector has two separate connection points. Or three, ....)

---

The comparison to resistors is only partly valid. Resistors have a lot less degrees of freedom. (only pitch, drill size, body dimensions)
Compare this to the degrees of freedom for a tab:
- pitch, drill size and body dimension are similar. (Here instead of body dimension i would use the connector blade dimensions. As already suggested above. The pitch should be included above.)
- In addition we also might need: overall body dimensions (at least height and probably overall width including the fins) maybe fin positions
- orientation (horizontal, 45 degree, ...)
- some have multiple connection points in one package.
- There are such connectors that include a plastic housing. (This again would need dimensions)

In short yes we can use a generic naming scheme but if we want it to include all possibilities it might get quite unwieldy. (I doubt a normal user might be able to find their correct footprint this way. Searching for MPN is much easier.)

**Comment by evanshultz** [comment]
@poeschlr 
To start, let me ground this conversation and try to add more supporting info instead of only experience and subjectivity. There will be some of that, but I've spent a bit of time to look around and see if my viewpoint is corroborated or not.

Tab connectors are a category of connector which has a few common ones but lots of diversity in the "noise". I feel this is similar to phono connectors, barrel jack connectors, and various other categories of connectors. KiCad has a number of symbols which represent these categories, even if not every symbol and not every footprint is available. I believe tab connectors are similar, so I'm quite surprised to see so much pushback. What's the reason *not* to support a common category of connectors both logically (symbol) and physically (footprint)?

Certainly it is true that while a single pin will provide all the necessary connectivity, having a symbol that reflects the style of the tab connector is very common in my experience. I'm OK with a single pin, but the important part to me is capturing the concept that this symbol is for a tab (the square symbol) as it helps the schematic to be self-documenting. Here are a variety of schematics which use a tab connector, and it seems to me that the ones using a non-specific symbol are less clear (and sometimes use text to explain):
- http://www.loneoceans.com/labs/sales/ud2precharge/SchematicUD2Prechargev10.png
- http://3.bp.blogspot.com/-d6gT8bDP2Gs/VJgwJ06aY3I/AAAAAAABGv4/-TcOb97JTPs/s1600/SCHEMATIC.bmp.jpg
- https://media.digikey.com/photos/rdl/EVL150W_ADP_SR_Schematic_Full.png
- https://fccid.io/img.php?id=526941&img=bg34.png
- http://4.bp.blogspot.com/-jgsidJurkFI/Vhyceh3k7JI/AAAAAAAAC2k/jEkBDEgWGqs/s1600/Main_sch.PNG
- http://schematic.narod.ru/Files/QSC2/cx302.pdf (page 5)

Having a generic tab symbol would also allow a custom FPfilter which only listed tab footprints in CvPcb, making it easier for KiCad users to select the footprint they desire. Those footprints could be generic or specific, depending on the FPfilter.

Your comment about other connectors is hype. I'm only talking about the specific tab connector because, at least in my experience, it's very common. That's why a dedicated symbol makes sense to me. If another category of connector warranted a unique symbol, of course discussion is open.

I disagree about the ability of a generic tab footprint to work. There's no need to cover every single possible type of tab connector if there are, say, just a handful that will cover 90% of the KiCad users. Only vertical connectors in the following blade widths are common IMHO: 0.110", 0.187", 0.205", and 0.250". I dug around a bit at Digi-Key and those 4 parts are the only common ones based on quantity of available parts, with a blade width of 0.032" (and thus still supporting thinner blades) being most common. Those are also the ones I would consider "generic", and having those 4 footprints would allow the vast majority of vertical tab parts to be supported.

Because I often use/see this type of connector, I may be closer to it and see it as an overly important. Many project would not use a tab connector, but most projects only use a small percentage of the library parts anyway.

@chmorgan My apologies for further derailing your PR. Thanks for your contribution to a component category that I think is important and is missing in KiCad.

**Comment by poeschlr** [comment]
@evanshultz what is your experience with traces below the stabilizing fins? Would it be a good idea to have a keepout area there? (Can many connection/disconnection cycles have the effect that the solder mask or even the trace gets damaged?)

In your research, how equal are these parts in other things then the connection side? 
- Have all of them only two pins? 
- Is the pin pitch standardized? (or at least limited to a small amount per connector type)
- What about the stabilizing fins, are they fairly standardized? (placement, size, ...)
- Is the over all height at least kind of standardized?

I am not opposed in having a general footprint. But if there is a lot of subtle variation it might get quite hard for a user to find the correct footprint for their connector.

**Comment by evanshultz** [comment]
@poeschlr 
They do vary, but I find there's an envelope that would work for most, if not all, vertical tabs. 3mm total width will cover the vast majority, while 4mm is conservative. It's too bad that KiCad doesn't support a real keepout tied to the DRC engine, though. Something like this:
image

Other questions:
- One is far less common, and for generic parts I only two-pin to be worthwhile.
- Yes. 5.00mm or 5.08mm.
- Above.
- There seems to be a reasonable range of common heights, from 6.35mm to ~8mm.

One other thing I've found is that AI machines may need different drill sizes than hand-inserted tabs. At least if you want the tabs to hold straight up through wave soldering. But since that can be changed in Pcbnew I suppose using 1.5mm holes will be fine for generic footprints.

Again, there are a lot of variations in tabs, but I can see 4 generic footprints and 1 symbol (which could also use any specific footprint) covering the vast majority of tabs.

**Comment by poeschlr** [comment]
@evanshultz or @chmorgan if any one of you two is prepared to take care of this please open a new pull request over at the [new footprint repo](https://github.com/KiCad/kicad-footprints). This one will be declared legacy and archived for this reason.

I would suggest we add a new Connector_Tab lib. I am prepared to add both generic footprints as well as specialized footprints to that lib as long as it is clear to users what footprints are generic/specialized. (Generic where it makes sense, specialized for everything else.)
